README.md

```diff
@@ -2,25 +2,64 @@
 
 # operatorkit
 
-The operatorkit package implements an opinionated framework for developing
-Kubernetes operators. It can be used as a library in golang. It is built on
-top of our [microkit](https://github.com/giantswarm/microkit) framework which
-provides base functionality like logging and error handling.
+The operatorkit is a library for creating [Kubernetes operators][operators]. It
+emerged as extracted common functionality of number of the operators we
+developed in Giant Swarm. The goal of the library is to provide common
+structure of the operator projects and encapsulate best practices learned
+during running operators in production.
+
+## Features
+
+- Reducing boilerplate code.
+- CRD/TPR primitives allowing to reliably create, watch custom resources, and
+  decode custom objects.
+- Independent packages. It is possible to use only a part of the library. 
+- Possibility to change behaviour that often is specific to an organization
+  like logging and error handling.
+- Minimal set of dependencies.
 
 ## Current Scope
 
-The initial scope is intentionally small and will be expanded as the community
-gains more experience developing operators.
+Project is split into independent packages providing complementary
+functionalities allowing to create production grade Kubernetes operators.
 
-* Kubernetes clientset using [client-go](https://github.com/kubernetes/client-go) for accessing the K8s APIs.
+- client - provides unified way of creating Kubernetes clients required by
+  other packages.
+- crd/tpr - provide CRD/TPR primitives, allowing to reliably create custom
+  resources, wait for their initialization, and generate endpoints URLs. 
+- informer - provides well defined watching functionality for virtually any
+  Kubernetes resource. It also provides a custom objects decoding functionality
+  reducing error prone boilerplate.
+- framework - provides a framework aiming to help writing reliable, robust
+  reconciliation loops.
```

Comment:
I would like to mention somehow that it resources are reconciled and that the implementation of the resource interface guides you step by step towards a proper reconciliation that drives the current state towards the desired state.

Reply:
This is what I wrote:

```
- framework - provides a framework aiming to help writing reliable, robust
  reconciliation loops. The heart of the framework is a Resouce interface
  guiding the user step by step towards a proper reconciliation that drives the
  current state towards the desired state. The Resource is designed to be
  composable. The framework provides useful Resource wrappers, making it easy
  to retry on failures, emit metrics and log consistent diagnostic messages.
```

WDYT?

Reply:
I like this.